### PR TITLE
#22857 tree view

### PR DESCRIPTION
**Pull request by danchr**
This PR contains the Liquor Tree implementation of the tree view. Due to being based off the org mapping branch, it also contains some of that work, but I've hidden the main button for now.

https://redmine.magenta-aps.dk/issues/22857

- [x] Have you updated the documentation?
- [x] Have you performed a smoke test of the application?
- [x] Have you written tests for your changes?
- [x] Have you updated the release notes?

### CODE REVIEW

File: backend/mora/service/orgunit.py [hunk 1]
```diff
@@ -415,6 +422,121 @@ def get_children(type, parentid):
...
+@blueprint.route('/ou/<uuid:unitid>/tree')
+@util.restrictargs('at')
+def get_unit_tree(unitid):
+    '''Obtain the unit hierarchy up to and including the given unit.
+
+    .. :quickref: Unit; Tree
+
+    :param unitid: The UUID of the organisational unit to be terminated.
+
+    :param see: http:get:`/service/ou/(uuid:unitid)/`.
```
Comment (cmoesgaard) [inline backend/mora/service/orgunit.py]: There's no parameter called `see` :slightly_smiling_face:

File: backend/mora/cli/base.py [hunk 2]
```diff
@@ -253,85 +253,77 @@ def full_run(**kwargs):
...
+    )
+
+    lora_port = lora_server.socket.getsockname()[1]
 
     with \
             test_support.psql() as psql, \
             mock.patch('settings.LOG_AMQP_SERVER', None), \
             mock.patch('settings.DB_HOST', psql.dsn()['host'], create=True), \
-            mock.patch('settings.DB_PORT', psql.dsn()['port'], create=True):
+            mock.patch('settings.DB_PORT', psql.dsn()['port'], create=True), \
```
Comment (agger-magenta) [inline backend/mora/cli/base.py]: Backslash continuations er ONDE! Men ja, der er vist ikke andet at gøre her.

File: backend/mora/exceptions.py [hunk 3]
```diff
@@ -40,6 +40,10 @@ def description(self):
     def code(self):
         return self.value[0]
 
+    def __call__(self, *args, **kwargs):
+        '''Raise an :py:class:`HTTPException` for this error code'''
```
Comment (agger-magenta) [inline backend/mora/exceptions.py]: Nice touch.

File: backend/mora/service/orgunit.py [hunk 4]
```diff
@@ -415,6 +413,172 @@ def get_children(type, parentid):
...
+                )
+            ),
+            key=lambda u: locale.strxfrm(u[mapping.NAME]),
+        )
+
+    # finally, return the root unit
+    return flask.jsonify(ancestors[id_path[0]])
+
+
+def get_unit_tree(c, orgid, unitids, with_siblings=False):
```
Comment (cmoesgaard) [inline backend/mora/service/orgunit.py]: `with_siblings` is never used AFAICT.

File: backend/mora/service/orgunit.py [hunk 5]
```diff
@@ -480,16 +644,13 @@ def get_orgunit(unitid):
...
-            org_unit_uuid=unitid,
-        )
+        exceptions.ErrorCodes.E_ORG_UNIT_NOT_FOUND(org_unit_uuid=unitid)
 
     return flask.jsonify(r)
 
 
 @blueprint.route('/o/<uuid:orgid>/ou/')
-@util.restrictargs('at', 'start', 'limit', 'query')
+@util.restrictargs('at', 'start', 'limit', 'query', 'tree')
```
Comment (cmoesgaard) [inline backend/mora/service/orgunit.py]: Add documentation for `tree`

Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: Also, is the `tree` parameter used anywhere in the calls from the frontend code?

Comment (danchr) [reply backend/mora/service/orgunit.py]: > Also, is the `tree` parameter used anywhere in the calls from the frontend code?

Not yet; the parameter allows for rendering a filtered tree, but the frontend does do that right now.

Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: > Not yet; the parameter allows for rendering a filtered tree, but the frontend doesn't do that right now.

Do we plan to? I'm just trying to find out why this functionality was added if it serves no purpose right now.

Comment (danchr) [reply backend/mora/service/orgunit.py]: > > Not yet; the parameter allows for rendering a filtered tree, but the frontend doesn't do that right now.
> 
> Do we plan to? I'm just trying to find out why this functionality was added if it serves no purpose right now.

I hope so; it'll make finding units much easier for both the mapping UI, and other uses in general.

Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: > > > Not yet; the parameter allows for rendering a filtered tree, but the frontend doesn't do that right now.
> > 
> > 
> > Do we plan to? I'm just trying to find out why this functionality was added if it serves no purpose right now.
> 
> I hope so; it'll make finding units much easier for both the mapping UI, and other uses in general.

I'm not a terribly big fan of adding code just for the sake of adding it. I'd rather we wait, and add this when there's an actual need for it.

**Pull request merged**